mitiq/pec/types/test_types.py

```diff
@@ -0,0 +1,249 @@
+# Copyright (C) 2020 Unitary Fund
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import numpy as np
+import pytest
+
+import cirq
+
+from mitiq.utils import _equal
+from mitiq.pec.types import NoisyOperation
+
+
+def test_init_with_gate():
+    ideal_gate = cirq.Z
+    real = np.array([[0.94, 0.05], [0.13, -0.98]])
+    noisy_op = NoisyOperation(ideal_gate, real)
+    assert isinstance(noisy_op.ideal_circuit, cirq.Circuit)
+    assert _equal(
+        noisy_op.ideal_circuit,
+        cirq.Circuit(ideal_gate.on(cirq.LineQubit(0))),
+        require_qubit_equality=False,
+    )
+    assert noisy_op.qubits == (cirq.LineQubit(0),)
+    assert np.allclose(noisy_op.ideal_matrix, cirq.unitary(cirq.Z))
+    assert np.allclose(noisy_op.real_matrix, real)
+    assert noisy_op.real_matrix is not real
+
+
+@pytest.mark.parametrize(
+    "qubit",
+    (cirq.LineQubit(0), cirq.GridQubit(1, 2), cirq.NamedQubit("Qubit")),
+)
+def test_init_with_operation(qubit):
+    ideal_op = cirq.H.on(qubit)
+    real = np.array([[0.51, 0.49], [0.46, -0.45]]) / np.sqrt(2)
```

Comment:
Here and for all the tests below, the size of the real matrix should be d^2 x d^2 instead of d x d.

Reply:
Maybe you could use: 
`real = np.eye(4)` (this is for sure a valid CPTP operation :-) ) 
or
`real = np.kron([[0.51, 0.49], [0.46, -0.45]], [[0.51, 0.49], [0.46, -0.45]]) `